### PR TITLE
Add number normalization test and address followup for getJsonObject

### LINKED ISSUE #10646: [FOLLOWUP] Query max instructions length in getJsonObject from JNI
Nit: The max length value should be queried from JNI so we always have such value sync. This can be addressed in the follow up work (which requires JNI to expose such value).

_Originally posted by @ttnghia in https://github.com/NVIDIA/spark-rapids/pull/10581#discussion_r1541941400_

Also some test cases need to clean up, see #10581 's comments.

### LINKED ISSUE #10688: [test] [follow-up] add IT to test number normalization for get-json-obj function
**Is your feature request related to a problem? Please describe.**
Add a IT as described in this link
[A clear and concise description of what the problem is. Ex. I wish the RAPIDS Accelerator for Apache Spark would [...]
](https://github.com/NVIDIA/spark-rapids-jni/pull/1944#issuecomment-2048966889)

```python
def json_string_to_float(x):
    if x == '"-Infinity"':
        return float('-inf')
    elif x == '"Infinity"':
        return float('inf')
    else:
        return float(x)

@pytest.mark.parametrize('data_gen', [StringGen('-?[1-9]\d{0,20}\.\d{1,20}', nullable=False),
                                      StringGen('-?[1-9]\d{0,5}\.\d{1,20}', nullable=False),
                                      StringGen('-?[1-9]\d{0,20}\.\d{1,5}', nullable=False),
                                      StringGen('-?[1-9]\d{0,5}\.\d{1,5}', nullable=False),
                                      StringGen('-?[1-9]\d{0,20}E-?\d{1,20}', nullable=False),
                                      StringGen('-?[1-9]\d{0,5}E-?\d{1,20}', nullable=False),
                                      StringGen('-?[1-9]\d{0,20}E-?\d{1,5}', nullable=False),
                                      StringGen('-?[1-9]\d{0,5}E-?\d{1,5}', nullable=False)], ids=idfn)
def test_get_json_object_floating_normalization(data_gen):
    schema = StructType([StructField("jsonStr", StringType())])
    gpu_res = [[row[1]] for row in with_gpu_session(
        lambda spark: unary_op_df(spark, data_gen, length=1000000).selectExpr(
            'a',
            'get_json_object(a,"$")'
            ).collect(),
        conf={'spark.rapids.sql.expression.GetJsonObject': 'true'})]
    cpu_res = [[row[1]] for row in with_cpu_session(
        lambda spark: unary_op_df(spark, data_gen, length=1000000).selectExpr(
            'a',
            'get_json_object(a,"$")'
            ).collect())]
    for i in range(len(gpu_res)):
        if not math.isclose(json_string_to_float(gpu_res[i][0]), json_string_to_float(cpu_res[i][0])):
            print("GPU: ", gpu_res[i][0], json_string_to_float(gpu_res[i][0]), " CPU: ", cpu_res[i][0], json_string_to_float(cpu_res[i][0]))
        assert math.isclose(json_string_to_float(gpu_res[i][0]), json_string_to_float(cpu_res[i][0]))

```

### PR DESCRIPTION
**Pull request by thirtiseven**
closes #10688 
closes #10646 

A minor follow-up pr for getJsonObject

This pr:
- Query max instructions length from JNI instead of hardcoding in plugin code.
- Adds tests for floating number normalization, it was fixed but didn't add cases in plugin.

### CODE REVIEW

File: integration_tests/src/main/python/get_json_test.py [hunk 1]
```diff
@@ -391,3 +391,31 @@ def test_get_json_object_number_normalization_legacy():
...
+    cpu_res = [[row[1]] for row in with_cpu_session(normalization)]
+    def json_string_to_float(x):
+        if x == '"-Infinity"':
+            return float('-inf')
+        elif x == '"Infinity"':
+            return float('inf')
+        else:
+            return float(x)
+    for i in range(len(gpu_res)):
+        assert math.isclose(json_string_to_float(gpu_res[i][0]), json_string_to_float(cpu_res[i][0]))
```
Comment (res-life) [inline integration_tests/src/main/python/get_json_test.py:421]: Nit: Add a comment like:
verify relatively diff < 1.0e-7

**Comment by res-life** [comment]
How long this test case will run?  < one minute?

**Review by res-life — APPROVED** [review]
LGTM

**Comment by thirtiseven** [comment]
> How long this test case will run? < one minute?

Yes. We used to run a case locally with a very large length, so it was slow, now it run fast with default length.

**Comment by thirtiseven** [comment]
build

**Review by res-life — APPROVED** [review]

**Pull request merged**